spec/README.md

```diff
@@ -0,0 +1,66 @@
+# Atomistics
+The `atomistics` package provides "interfaces for  atomistic simulation codes and workflows". While the 
+[user documentation](https://atomistics.readthedocs.io/) already covers how to use the `atomistics` package, this 
+document explains the motivation and vision of the development the `atomistics` package, the underlying concepts and 
+the resulting internal structure of the `atomistics` package. 
+
+## Vision 
+The `atomistics` package provides "interfaces for  atomistic simulation codes and workflows".
+
+To calculate a physical property with an atomistic simulation code, there are typically two ways: 
+
+* **internal implementation**: The simulation code already provides the internal functionality to calculate the physical
+  property. Most atomistic simulation codes can calculate, potential energies and forces for a given atomistic 
+  structure. Some simulation codes already provide internal functionality to calculate more complex physical properties, 
+  like the phonon density of states or energy barriers using the nudged elastic band method. 
+* **external tools**: The second approach is to use a workflow, which combines a series of individual calculation of 
+  atomistic simulation codes to calculate physical properties. These workflows can be simple shell scripts for example 
+  for the calculation of energy volume curves or more complex physical properties, like the phonon density introduced 
+  above. 
+
+The challenge for the `atomistics` package is balancing both approaches. The first is commonly computationally more 
+efficient, while the second allows to directly compare different simulation codes and simulation methods. The 
+`atomistics` packages provides an abstraction to balance between both approaches. By default, it uses the internal 
+implementation when available and falls back to a universal python based implementation when necessary. With this 
+approach the `atomistics` package allows users to calculate a wide-range of physical properties with a wide-range of 
+simulation codes. 
```

Comment:
I find it very nice that it addresses a main problem here. However, it doesn't really address how we approach it. For example, based on this description I would not know where to go if I have my own atomistic code. I think it's important that we refer to the output classes at least. In addition to this, we would restructure the input classes to have a similar modularity, and the interactive mode has to be conceptually somehow included. This being said, it's not a problem of this document, but it's more a fundamental problem of how this module is structured, so I'm also ok with not mentioning it here right away.